src/main/scala/chisel3/annotations/annotations.scala

```diff
@@ -0,0 +1,45 @@
+// SPDX-License-Identifier: Apache-2.0
+
+package chisel3.annotations
+
+import firrtl.annotations._
+import firrtl.FirrtlUserException
+
+/** The lowering convention of a module
+  * Controls how a module's ports are lowered to verilog.
+  */
+sealed class Convention(val string: String) {
+  def serialize: String = string
+}
+
+object Convention {
+
+  /** Undefined internal convention */
+  case object Internal extends Convention("internal")
+
+  /** Fully scalarize aggregate ports */
+  case object Scalarized extends Convention("scalarized")
+
+  def deserialize(string: String): Convention = string match {
+    case "internal"   => Internal
+    case "scalarized" => Scalarized
+    case _            => throw new FirrtlUserException(s"Unrecognized Convention: $string")
+  }
+}
+
+/** Annotation to force the compiler to use a specified port convention.
+  * May only be applied to modules.
+  */
+case class ConventionAnnotation(target: ModuleName, convention: String) extends SingleTargetAnnotation[ModuleName] {
+  def duplicate(target: ModuleName): ConventionAnnotation =
+    ConventionAnnotation(target, convention)
+}
+
+object ConventionAnnotation {
+
+  /** Create a ConventionAnnotation
+    * Helper to convert create an annotation from the Convention enum.
+    */
+  def apply(target: ModuleName, convention: Convention): ConventionAnnotation =
+    ConventionAnnotation(target, convention.serialize)
+}
```

Comment:
This all looks reasonable, but this does need some Chisel user-API, some function-like API that accepts a Chisel module as an argument and annotates it. See `dontTouch` for an example (although that operates on `Data` instead of `RawModule`): https://github.com/chipsalliance/chisel3/blob/3c14563743971842069679c263adbc1aa74adc3c/core/src/main/scala/chisel3/dontTouch.scala

Reply:
Done, thanks!